File: web-server/plugins/slycat-model-wizards/slycat-model-wizards.py
# Copyright (c) 2013, 2018 National Technology and Engineering Solutions of Sandia, LLC . Under the terms of Contract
# DE-NA0003525 with National Technology and Engineering Solutions of Sandia, LLC, the U.S. Government
# retains certain rights in this software.

def register_slycat_plugin(context):
  import os
  context.register_wizard("slycat-delete-model", "Delete Model", require={"action":"delete", "context":"model"})
  context.register_wizard("slycat-edit-model", "Model Name & Info", require={"action":"edit", "context":"model"})

  context.register_wizard("slycat-edit-project", "Edit Project", require={"action":"edit", "context":"model"})
  
  context.register_wizard("slycat-create-template", "Template", require={"action":"create", "context":"model"})
  # Alex commenting this out and instead using the line below because the Save As functionality is not ready yet.
  # context.register_wizard("slycat-create-saved-bookmark", "Save As", require={"action":"create", "context":"model"})
  context.register_wizard("slycat-create-saved-bookmark", "Create New", require={"action":"create", "context":"model"})
  context.register_wizard("slycat-apply-template", "Apply Template", require={"action":"edit", "context":"model"})
  context.register_wizard("slycat-reset-model", "Reset Model", require={"action":"edit", "context":"model"})
  context.register_wizard("slycat-info-model", "Model Details", require={"action":"info", "context":"model"})

File: web-server/plugins/slycat-model-wizards/slycat-model-wizards.js
import ko from "knockout";

import slycat_edit_model from 'plugins/slycat-model-wizards/edit-ui';
import slycat_delete_model from 'plugins/slycat-model-wizards/delete-ui';
import slycat_apply_template from 'plugins/slycat-model-wizards/apply-template-ui';
import slycat_create_saved_bookmark from 'plugins/slycat-model-wizards/create-saved-bookmark-ui';
import slycat_create_template from 'plugins/slycat-model-wizards/create-template-ui';
import slycat_info_model from 'plugins/slycat-model-wizards/info-ui';
import slycat_reset_model from 'plugins/slycat-model-wizards/reset-ui';
import slycat_edit_project from 'plugins/slycat-project-wizards/edit-project-ui';

ko.components.register('slycat-edit-project', slycat_edit_project);
ko.components.register('slycat-edit-model', slycat_edit_model);
ko.components.register('slycat-delete-model', slycat_delete_model);
ko.components.register('slycat-apply-template', slycat_apply_template);
ko.components.register('slycat-create-saved-bookmark', slycat_create_saved_bookmark);
ko.components.register('slycat-create-template', slycat_create_template);
ko.components.register('slycat-info-model', slycat_info_model);
ko.components.register('slycat-reset-model', slycat_reset_model);